### PR TITLE
qemu_img_check_fragmentation: new case to check file fragmentation

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1881744

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_check_fragmentation.q35: PASS (54.80 s)

**Comment by XueqiangWei** [comment]
@zhencliu please help review it, many thanks.

File: qemu/tests/cfg/qemu_img_check_fragmentation.cfg [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    virt_test_type = qemu
+    type = qemu_img_check_fragmentation
+    required_qemu = [5.1.0-2, )
+    start_vm = no
+    force_create_image_stg = yes
+    images = stg
+    image_size_stg = 10G
+    image_format_stg = raw
+    image_name_stg = "images/stg"
+    offset = "0 4096"
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_check_fragmentation.cfg:12]: offsets = xxx ?

File: qemu/tests/cfg/qemu_img_check_fragmentation.cfg [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    timeout = 900
+    fragmentation_maximum = 10000
+    check_fragmentation_cmd = "filefrag %s"
+    extents_number = ":\s+(\d+)\s+extents found"
```
Comment (zhencliu) [inline qemu/tests/cfg/qemu_img_check_fragmentation.cfg:16]: extents_number_pattern?

Comment (XueqiangWei) [reply qemu/tests/cfg/qemu_img_check_fragmentation.cfg:16]: Agree with you,  thanks.

File: qemu/tests/qemu_img_check_fragmentation.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+def run(test, params, env):
+    """
+    Check file fragmentation.
+    1. Create a raw image with 10G.
+    2. Create a badly fragmented file with qemu-img bench.
+    3. Check file fragmentation. The extents should less than 10000.
+       With 1 MB extents, the theoretical maximum for a 10 GB image
+       is 10000 extents (10000 * 1 MB = 10 GB)
```
Comment (zhencliu) [inline qemu/tests/qemu_img_check_fragmentation.py:16]: we don't use a factor 1024 here?

Comment (XueqiangWei) [reply qemu/tests/qemu_img_check_fragmentation.py:16]: it's developer's comment. I will update it. thanks

File: qemu/tests/cfg/qemu_img_check_fragmentation.cfg (hunk 1 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/qemu_img_check_fragmentation.cfg:12]: I will update it, thanks.

**Comment by leidwang** [comment]
Test this patch on my host, result are as follows:

(1/1) Host_RHEL.m8.u4.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_check_fragmentation.q35: PASS (272.71 s)

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m8.u3.product_av.raw.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.qemu_img_check_fragmentation.q35: PASS (54.63 s)

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**